### PR TITLE
[SW-1704] Expose Extra Http Headers for H2O Nodes

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: doc/src/site/sphinx/configuration/configuration_properties.rst [hunk 1]
```diff
@@ -146,19 +146,19 @@ Configuration properties independent of selected backend
 |                                                    |                | corresponding parameters in Sparkling  |
 |                                                    |                | Water.                                 |
 +----------------------------------------------------+----------------+----------------------------------------+
-| **H2O client parameters**                          |                |                                        |
-+----------------------------------------------------+----------------+----------------------------------------+
-| ``spark.ext.h2o.client.flow.dir``                  | ``None``       | Directory where flows from H2O Flow    |
-|                                                    |                | are saved.                             |
-+----------------------------------------------------+----------------+----------------------------------------+
-| ``spark.ext.h2o.client.flow.extra.http.headers``   | ``None``       | Extra HTTP headers that will be used   |
```
Comment (mn-mikke) [inline doc/src/site/sphinx/configuration/configuration_properties.rst]: @jakubhava Any idea how to properly deprecate `spark.ext.h2o.client.flow.extra.http.headers`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 2]
```diff
@@ -101,3 +101,8 @@ without the need of releasing H2O if there is only change on the Sparkling Water
 ``SWPatchVersion``. The new version therefore looks, for example, like ``3.26.0.9-2-2.4``. This version tells us this
 Sparkling Water is integrating H2O ``3.26.0.9``, it is the second release with ``3.26.0.9`` version and is for Spark ``2.4``.
 
+Renamed Property for Passing Extra HTTP Headers for Flow UI
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Since Flow UI can also run on each H2O node, we decided to rename the configuration property ``spark.ext.h2o.client.flow.extra.http.headers``
+to ``spark.ext.h2o.flow.extra.http.headers``.
```
Comment (jakubhava) [inline doc/src/site/sphinx/migration_guide.rst]: Thank you! I would stress out that it means that the configuration is applied to worker & client nodes

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 1 above)
Comment (jakubhava) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: We can use https://github.com/h2oai/sparkling-water/blob/6fa71b8053630afd931b1abbba018cec70e92275/core/src/main/scala/org/apache/spark/h2o/H2OConf.scala#L116, what do think?

File: doc/src/site/sphinx/migration_guide.rst (hunk 2 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/migration_guide.rst]: ok, will do. Thanks!

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 1 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: That's smart, thanks! Will deprecate it in another PR.

**Comment by jakubhava** [comment]
Note: this PR can remove the deprecation warning in https://github.com/h2oai/sparkling-water/pulls

**Pull request merged**